ffi/src/lib.rs

```diff
@@ -388,14 +526,44 @@ fn create_client_internal(
             let redis_error = err.into();
             errors::error_message(&redis_error)
         })?;
+
+    let is_subscriber = request.pubsub_subscriptions.is_some() && pubsub_callback as usize != 0;
+    let (push_tx, mut push_rx) = tokio::sync::mpsc::unbounded_channel();
+    let tx = match is_subscriber {
+        true => Some(push_tx),
+        false => None,
+    };
+
     let client = runtime
-        .block_on(GlideClient::new(ConnectionRequest::from(request), None))
+        .block_on(GlideClient::new(ConnectionRequest::from(request), tx))
         .map_err(|err| err.to_string())?;
+
+    // Create the client adapter that will be returned and used as conn_ptr
     let core = Arc::new(CommandExecutionCore {
         client,
         client_type,
     });
-    Ok(ClientAdapter { runtime, core })
+    let client_adapter = Arc::new(ClientAdapter { runtime, core });
+    // Clone client_adapter before moving it into the async block
+    let client_adapter_ptr = Arc::as_ptr(&client_adapter).addr();
+
+    // If pubsub_callback is provided (not null), spawn a task to handle push notifications
+    if is_subscriber {
+        client_adapter.runtime.spawn(async move {
+            while let Some(push_msg) = push_rx.recv().await {
+                if push_msg.kind == redis::PushKind::Message
+                    || push_msg.kind == redis::PushKind::PMessage
+                    || push_msg.kind == redis::PushKind::SMessage
+                {
+                    unsafe {
+                        process_push_notification(push_msg, pubsub_callback, client_adapter_ptr);
+                    }
+                }
+            }
+        });
```

Comment:
1. Move to a dedicate function (e.g. push_notification_loop) 
2. Currently you're silently ignoring all other push notifications. Please log when pushed notification is ignored and add a TODO to handle other push kinds as well (e.g. Invalidate, Other) & open an issue so we won't forget it.